File: solutions/swift/leap/Sources/Leap/Leap.swift
class Year {
  private var year: Int
  init(calendarYear: Int) { year = calendarYear }
  var isLeapYear: Bool {
    year.isMultiple(of: 4) && !year.isMultiple(of: 100) || year.isMultiple(of: 400)
  }
}

File: solutions/vbnet/leap/Leap.vb
Public Module Leap
    Public Function IsLeapYear(ByVal year As Integer) As Boolean
        Dim IsDivisibleBy = 
            Function(n As Integer) As Boolean
                Return (year Mod n) = 0
            End Function
        Return IsDivisibleBy(4) AndAlso 
            Not IsDivisibleBy(100) OrElse 
            IsDivisibleBy(400)
    End Function
End Module